File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        b, s = 0, 1

        while s < len(prices):
            if prices[s] <= prices[b]:
                b = s
            if prices[s] > prices[b]:
                profit = max(profit, prices[s] - prices[b])
            s = s + 1

        return profit

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cpp
#include <algorithm>

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res = 0;
        int l = 0;
        vector<char> sub_array;

        for (int r = 0; r < s.size(); r++) {
            while (std::find(sub_array.begin(), sub_array.end(), s[r]) != sub_array.end()) {
                std::erase(sub_array, s[l]);
                l++;
            };
            sub_array.push_back(s[r]);
            res = std::max(res, (int)sub_array.size());
        };

    return res;
    
    };
};
